agent/agent.go

```diff
@@ -87,14 +87,19 @@ func hostProxy(ctx context.Context, host, shimPath string, injectShimCode bool)
 	})
 	hostProxy.FlushInterval = 100 * time.Millisecond
 	var h http.Handler = hostProxy
+	h = sessionLRU.SessionHandler(h)
 	if shimPath != "" {
 		var err error
-		h, err = websockets.Proxy(ctx, hostProxy, host, shimPath, injectShimCode, *rewriteWebsocketHost)
+		h, err = websockets.Proxy(ctx, h, host, shimPath, *rewriteWebsocketHost, sessionLRU.SessionHandler)
+		if injectShimCode {
+			hostProxy.ModifyResponse = func(resp *http.Response) error {
+				return websockets.ShimBody(resp, shimPath)
```

Comment:
This changes it so that instead of the template being compiled once at startup it is recompiled during the processing of every response.

Let's change this so that the `ShimBody` method returns a function so this block can be:

```go
			hostProxy.ModifyResponse = websockets.ShimBody(shimPath)
```

Reply:
Done